### PR TITLE
Ack rule endpoint

### PR DESCRIPTION
**Pull request by tisnik**
# Description

The first (of five) new `ack` endpoints implementation. This one is used to "ack" a rule. If the rule is already acked, different HTTP code is returned:

#### Ack new rule

Request to the service:

```
curl -v -X POST -H "Content-Type: application/json" --data '{"rule_id":"foo|bar", "justification":"xyzzy"}' "localhost:8080/api//v2/ack"
```

Response from the service:

```
< HTTP/1.1 201 Created
< Content-Type: application/json; charset=utf-8
< Date: Sun, 05 Sep 2021 14:29:33 GMT
< Content-Length: 168
< 
{
        "rule": "foo|bar",
        "justification": "xyzzy",
        "created_by": "onlineTester",
        "created_at": "2021-09-05T16:29:33+02:00",
        "updated_at": "2021-09-05T16:29:33+02:00"
}
```

#### Ack existing rule

Request to the service:

```
curl -v -X POST -H "Content-Type: application/json" --data '{"rule_id":"existing|rule", "justification":"xyzzy"}' "localhost:8080/api/v2/ack"
```

Response from the service:

```
< HTTP/1.1 200 OK
< Content-Type: application/json; charset=utf-8
< Date: Sun, 05 Sep 2021 14:35:51 GMT
< Content-Length: 168
< 
{
        "rule": "foo|bar",
        "justification": "xyzzy",
        "created_by": "onlineTester",
        "created_at": "2021-09-05T16:35:25+02:00",
        "updated_at": "2021-09-05T16:35:25+02:00"
}

```

Please note that `acks_utils.go` source file might look complicated, but these functions and method will be reused later.

Additionally, there are not unit tests prepared yet, as our "unit test framework" needs some improvements:

https://github.com/RedHatInsights/insights-results-smart-proxy/issues/638
https://github.com/RedHatInsights/insights-results-smart-proxy/issues/639

Also there's TODO to be done later (as we don't want to slow front-end work):
https://github.com/RedHatInsights/insights-results-smart-proxy/issues/640

## Type of change

- New feature (non-breaking change which adds functionality)
- Bump-up dependent library (no changes in the code)

## Testing steps

N/A at this moment, but can be tested against running Insights Results Aggregator (already performed of course)

## Checklist
* [ ] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: server/acks_handlers.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+//   "justification": "string",  <- can not be set by this call!!!
+//   "created_by": "string",
+//   "created_at": "2021-09-04T17:52:48.976Z",
+//   "updated_at": "2021-09-04T17:52:48.976Z"
+// }
+//
+// HTTP/1.1 200 OK is returned if rule has been already acked
+// HTTP/1.1 201 Created is returned if rule has been acked by this call
+func (server *HTTPServer) acknowledgePost(writer http.ResponseWriter, request *http.Request) {
+	const readRuleStatusError = "can not retrieve rule disable status from Aggregator"
```
Comment (juandspy) [inline server/acks_handlers.go:57]: This variable is unused.

File: server/acks_handlers.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+		return
+	}
+
+	// display parsed rule ID and error key
+	log.Info().
+		Str("ruleID", string(ruleID)).
+		Str("errorKey", string(errorKey)).
+		Msg("Parsed rule selector")
+
+	// test if the rule has been acknowledget already
```
Comment (juandspy) [inline server/acks_handlers.go:97]: Typo: acknowledged

**Review by juandspy — APPROVED** [review]
Just some nitpicks ;)

File: server/acks_handlers.go (hunk 2 above)
Comment (tisnik) [reply server/acks_handlers.go:97]: ah nice catch!

Comment (tisnik) [reply server/acks_handlers.go]: Resolved by new commit

File: server/acks_utils.go [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+	if err != nil {
+		handleServerError(writer, err)
+		return types.OrgID(0), types.UserID(""), err
+	}
+
+	// Organization ID and user ID are to be provided in the token
+	orgID := authToken.Internal.OrgID
+	userID := authToken.AccountNumber
+	return orgID, userID, nil
+}
```
Comment (joselsegura) [inline server/acks_utils.go:73]: I think that we already have some functions to extract the user and organization IDs

**Review by joselsegura — APPROVED** [review]

File: server/acks_handlers.go (hunk 1 above)
Comment (tisnik) [reply server/acks_handlers.go:57]: it's constant :) will recheck it

**Pull request merged**